core/dbt/task/rpc_server.py

```diff
@@ -1,16 +1,126 @@
 import json
-import os
+import multiprocessing
+import time
 
 from jsonrpc import Dispatcher, JSONRPCResponseManager
 
 from werkzeug.wrappers import Request, Response
 from werkzeug.serving import run_simple
 
-from dbt.logger import RPC_LOGGER as logger
+from dbt.logger import RPC_LOGGER as logger, add_queue_handler
 from dbt.task.base import ConfiguredTask
 from dbt.task.compile import CompileTask, RemoteCompileTask
 from dbt.task.run import RemoteRunTask
 from dbt.utils import JSONEncoder
+from dbt.compat import QueueEmpty
+import dbt.exceptions
+from dbt import rpc
+
+
+class RequestTaskHandler(object):
+    def __init__(self, task):
+        self.task = task
+        self.queue = None
+        self.process = None
+        self.started = None
+        self.timeout = None
+        self.logs = []
+
+    def _next_timeout(self):
+        if self.timeout is None:
+            return None
+        end = self.started + self.timeout
+        timeout = end - time.time()
+        if timeout < 0:
+            raise dbt.exceptions.RPCTimeoutException(self.timeout)
+        return timeout
+
+    def _wait_for_results(self):
+        """Wait for results off the queue. If there is a timeout set, and it is
+        exceeded, raise an RPCTimeoutException.
+        """
+        while True:
+            get_timeout = self._next_timeout()
+            try:
+                msgtype, value = self.queue.get(timeout=get_timeout)
+            except QueueEmpty:
+                raise dbt.exceptions.RPCTimeoutException(self.timeout)
+
+            if msgtype == 'log':
+                self.logs.append(value)
+            else:
+                return msgtype, value
+
+    def _join_process(self):
+        try:
+            msgtype, result = self._wait_for_results()
+        except dbt.exceptions.RPCTimeoutException as exc:
+            self.process.terminate()
+            raise rpc.timeout_error(self.timeout)
+        except dbt.exceptions.Exception as exc:
+            raise rpc.dbt_error(exc)
+        except Exception as exc:
+            raise rpc.server_error(exc)
+        finally:
+            self.process.join()
+
+        self.process = None
+        self.queue = None
```

Comment:
are these two lines significant from a GC perspective? if so it should probably go into the `finally` block

Reply:
Nope, they're not significant, I should just remove them.